ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -4025,7 +4025,9 @@ private OBLogicalPlanGenState beginGenOBLogicalPlan(
         // replace each of the position alias in ORDERBY with the actual column
         if (ref != null && ref.getToken().getType() == HiveParser.Number) {
           if (isObyByPos) {
-            fieldIndex = getFieldIndexFromColumnNumber(selectOutputRR, ref);
```

Comment:
is it ok that `selectOutputRR` is null?
these RR things are a bit messed up...I wanted to fix things in the vicinity in https://issues.apache.org/jira/browse/HIVE-24830; so I think that could become a source of problems...

Reply:
if you see no other hazards which may come out from that being `null` we can get this in - but is it true that if the RR is `null` - then the select is an identity?

Reply:
HIVE-24830 doesn't solve this.

When generating the Calcite plan of values queries the source of the `Order by` is a `HiveTableFunctionScan` instead of a `Project`.
The `values` clause is converted to a function call and as a result we end up with a `HiveTableFunctionScan`
https://github.com/apache/hive/blob/c39cc56e28172397d573ca9072f55551fef86a56/ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java#L4606

This case the output RR is null:
https://github.com/apache/hive/blob/c39cc56e28172397d573ca9072f55551fef86a56/ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java#L4870

Reply:
yes; I know that will not solve this - the general problem is that going to the `input` instead of the current one could be a mistake

Then I suppose we are getting the `outputRel` from [here](https://github.com/apache/hive/blob/c39cc56e28172397d573ca9072f55551fef86a56/ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java#L4794) can't we fix the RR of that UDTF node - instead of returning `null` there and falling back to the `input` here?

Reply:
Here is a different approach: return the output RR of the UDTF when transforming to `HiveTableFunctionScan`
Could you please review? #2899